rct229/rulesets/ashrae9012019/ruleset_functions/get_energy_required_to_heat_swh_use.py

```diff
@@ -0,0 +1,220 @@
+    drain_heat_recovery_efficiency = 0.0
+    supply_temperature = None
+    if use_units in VOLUME_BASED_USE_UNIT or is_heat_recovered_by_drain:
+        distribution_system_id = getattr_(
+            swh_use, "service_water_heating_uses", "served_by_distribution_system"
+        )
+        distribution_system = (
+            find_exactly_one_service_water_heating_distribution_system(
+                rmd, distribution_system_id
+            )
+        )
+        if use_units in VOLUME_BASED_USE_UNIT:
+            supply_temperature = getattr_(
+                distribution_system,
+                "service_water_heating_distribution_systems",
+                "design_supply_temperature",
+            )
+            inlet_temperature_schedule_id = getattr_(
+                distribution_system,
+                "service_water_heating_distribution_systems",
+                "entering_water_mains_temperature_schedule",
+            )
+            inlet_temperature_schedule = find_exactly_one_schedule(
+                rmd, inlet_temperature_schedule_id
+            )
+            inlet_temperature_hourly_values = getattr_(
+                inlet_temperature_schedule, "schedules", "hourly_values"
+            )
+        if is_heat_recovered_by_drain:
+            drain_heat_recovery_efficiency = getattr_(
+                distribution_system,
+                "service_water_heating_distribution_systems",
+                "drain_heat_recovery_efficiency",
+            )
+
+            assert_(
+                0.0 <= drain_heat_recovery_efficiency <= 1.0,
+                "`drain_heat_recovery_efficiency` value must be between 0 and 1.",
+            )
+
+    space_ids = get_spaces_served_by_swh_use(rmd, swh_use["id"])
+    space_within_building_segment_ids = find_all(
+        "$.zones[*].spaces[*].id", building_segment
+    )
+    spaces = [
+        find_exactly_one_space(rmd, space_id)
+        for space_id in space_ids
+        if space_id in space_within_building_segment_ids
+    ]
+
+    if not spaces and use_units not in REQUIRED_USE_UNIT:
+        spaces = find_all("$.zones[*].spaces[*]", building_segment)
+
+    hourly_values = (
+        getattr_(hourly_schedule, "hourly_schedule", "hourly_values")
+        if hourly_schedule is not None
+        else [1] * len(inlet_temperature_hourly_values)
+    )
+
+    swh_use_value = swh_use.get("use", 0.0)
+    energy_required_by_space = {}
+    volume = ZERO.VOLUME
+    for space in spaces:
+        space_id = space["id"]
+        if use_units == SERVICE_WATER_HEATING_USE_UNIT.POWER_PER_PERSON:
+            power = swh_use_value * ureg("W") * space.get("number_of_occupants", 0)
+            energy_required = (
+                power
+                * sum(hourly_values)
+                * ureg("hr")
+                * (1 - drain_heat_recovery_efficiency)
+            )
+            energy_required_by_space[space_id] = energy_required
```

Comment:
If the unit is power-based, the output `energy_required` is in the unit of W*h, how can I convert it into Btu? Could you take a look? Thanks @yunjoonjung-PNNL

Reply:
You can simply multiply the line 140 with ```.to("Btu")``` to ensure the outcome is in ```Btu``` unit.

Reply:
Technically, you don't have to do any conversion because Pint knows the units. For example, pint knows how to compare a `W*h` with a 'Btu'.